source/android/adaptivecards/src/main/java/com/microsoft/adaptivecards/renderer/input/ChoiceSetInputRenderer.java

```diff
@@ -0,0 +1,189 @@
+package com.microsoft.adaptivecards.renderer.input;
+
+import android.content.Context;
+import android.support.v4.app.FragmentManager;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.CheckBox;
+import android.widget.LinearLayout;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Spinner;
+
+import com.microsoft.adaptivecards.objectmodel.ChoiceInput;
+import com.microsoft.adaptivecards.objectmodel.ChoiceInputVector;
+import com.microsoft.adaptivecards.objectmodel.ChoiceSetStyle;
+import com.microsoft.adaptivecards.renderer.inputhandler.CheckBoxSetInputHandler;
+import com.microsoft.adaptivecards.renderer.inputhandler.ComboBoxInputHandler;
+import com.microsoft.adaptivecards.renderer.inputhandler.IInputHandler;
+import com.microsoft.adaptivecards.objectmodel.BaseCardElement;
+import com.microsoft.adaptivecards.objectmodel.ChoiceSetInput;
+import com.microsoft.adaptivecards.objectmodel.HostConfig;
+import com.microsoft.adaptivecards.renderer.BaseCardElementRenderer;
+import com.microsoft.adaptivecards.renderer.inputhandler.RadioGroupInputHandler;
+
+import java.util.List;
+import java.util.Vector;
+
+/**
+ * Created by bekao on 6/25/2017.
+ */
+
+public class ChoiceSetInputRenderer extends BaseCardElementRenderer
+{
+    private ChoiceSetInputRenderer()
+    {
+    }
+
+    public static ChoiceSetInputRenderer getInstance()
+    {
+        if (s_instance == null)
+        {
+            s_instance = new ChoiceSetInputRenderer();
+        }
+
+        return s_instance;
+    }
+
+    public View renderCheckBoxSet(
+            Context context,
+            ChoiceSetInput choiceSetInput,
+            Vector<IInputHandler> inputActionHandlerList,
+            HostConfig hostConfig)
+    {
+        LinearLayout layout = new LinearLayout(context);
+        layout.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
+        layout.setOrientation(LinearLayout.VERTICAL);
+
+        List<CheckBox> checkBoxList = new Vector<CheckBox>();
+        ChoiceInputVector choiceInputVector = choiceSetInput.GetChoices();
+        long size = choiceInputVector.size();
+        for (int i = 0; i < size; i++)
+        {
+            ChoiceInput choiceInput = choiceInputVector.get(i);
+            CheckBox checkBox = new CheckBox(context);
+            checkBox.setText(choiceInput.GetTitle());
+            checkBox.setChecked(choiceInput.GetIsSelected());
+            checkBoxList.add(checkBox);
+            layout.addView(checkBox);
```

Comment:
not sure I understand.

---
In reply to: [127286753](https://github.com/Microsoft/AdaptiveCards/pull/461#discussion_r127286753) [](ancestors = 127286753)